dulwich/porcelain.py

```diff
@@ -488,7 +488,7 @@ def init(path=".", *, bare=False, symlinks: Optional[bool] = None):
 
 def encode_refspecs(refspecs, refspec_encoding):
     if refspecs is None:
-        return [b"HEAD"]
+        return [b"HEAD", b"refs/"]
```

Comment:
This is a significant change in behaviour, as it doesn't just change what refs we discover but forces us to fetch all refs rather than just HEAD.

Reply:
With protocol v1, server-side ref filtering is non-existent.
With protocol v2, the client can ask the server to filter refs, but the server is free to return a non-filtered result. So in any case, the client needs to be prepared to filter refs on its own.

But indeed, I used this parameter to test the protocol v2 functionality. It was never intended to be used with protocol v1. I agree that having this parameter work consistently across all protocol versions would be much better.

The problem where Github does not return all refs unless we ask for a filter only affects fetches with protocol v2 over SSH. Protocol v2 over HTTP works fine.  I consider this to a bug in their git/ssh server. We're just trying to work around that.